src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -515,6 +518,28 @@ private static IEnumerable<IMSBuildItem> GetItemByType(IEnumerable<IMSBuildItem>
             return items.Where(e => type.Equals(e.GetProperty("Type"), StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Return the parsed version or 1.0.0 if the property does not exist.
+        /// </summary>
+        private static NuGetVersion GetVersion(IMSBuildItem item)
+        {
+            var versionString = item.GetProperty("Version");
+            NuGetVersion version = null;
+
+            if (string.IsNullOrEmpty(versionString))
+            {
+                // Default to 1.0.0 if the property does not exist
+                version = new NuGetVersion(1, 0, 0);
+            }
+            else
+            {
+                // Snapshot versions are not allowed in .NETCore
+                version = NuGetVersion.Parse(versionString);
```

Comment:
Is an exception desired if a bad version is passed? I think so (since we have other exception cases acting like this), but wanted to double check.